### PR TITLE
Add dynamic compass location support

### PR DESCRIPTION
**Pull request by Incompleteusern**

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/overlays/QuestInfoOverlayFeature.java [hunk 1]
```diff
@@ -138,6 +147,7 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
 
             if (toRender.get(0).getText() == null) {
                 // Set at first use; I18n is not available at initialization time
+                // TODO check if above comment still valid
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/QuestInfoOverlayFeature.java]: It is not. This comment was the reason features were made to init after i18n

File: common/src/main/java/com/wynntils/features/user/overlays/QuestInfoOverlayFeature.java [hunk 2]
```diff
@@ -61,14 +62,22 @@ public void onScoreboardSegmentChange(ScoreboardSegmentAdditionEvent event) {
...
 
-        Optional<Location> location = event.getQuestInfo().getNextLocation();
-        if (location.isEmpty()) return;
+        return location.get();
+    };
 
-        CompassModel.setCompassLocation(location.get());
+    @SubscribeEvent
+    public void onTrackedQuestUpdate(TrackedQuestUpdateEvent event) {
+        // set if valid
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/QuestInfoOverlayFeature.java:64]: when is it not valid?

File: common/src/main/java/com/wynntils/features/user/overlays/QuestInfoOverlayFeature.java [hunk 3]
```diff
@@ -61,14 +62,22 @@ public void onScoreboardSegmentChange(ScoreboardSegmentAdditionEvent event) {
         }
     }
 
-    @SubscribeEvent
-    public void onTrackedQuestUpdate(TrackedQuestUpdateEvent event) {
-        if (event.getQuestInfo() == null || !autoTrackQuestCoordinates) return;
+    private static final Supplier<Location> supplier = () -> {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/QuestInfoOverlayFeature.java]: Debatable if this should be stored here or in QuestManager. Let's move this to QuestManager and still set the compass here in the event?

File: common/src/main/java/com/wynntils/wynn/model/CompassModel.java [hunk 4]
```diff
@@ -5,39 +5,73 @@
...
+        if (supplier == null) return;
+
+        Location newLocation = supplier.get();
+
+        if (newLocation == null) { // drop location
+            reset();
+        } else if (compassLocation != newLocation) { // update location
+            compassLocation = newLocation;
+
+            if (McUtils.mc().level != null) {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/CompassModel.java:36]: I am pretty sure client tick only happens if there is a level. But we can keep this.

**Review by kristofbolyai — COMMENTED** [review]
IMO move the quest supplier to quest manager, then this is good to go

**Review by kristofbolyai — DISMISSED** [review]

**Comment by kristofbolyai** [comment]
Actually, @Incompleteusern this might not be the right step to make. Why not make Pois have a supplier for location? That way, you can turn Poi into an interface too, just like we discussed in discord earlier

File: common/src/main/java/com/wynntils/wynn/model/CompassModel.java [hunk 5]
```diff
@@ -5,39 +5,73 @@
...
 
             return Optional.of(waypointPoi);
         }
 
         return Optional.empty();
     }
 
-    public static void setCompassLocation(Location compassLocation) {
-        CompassModel.compassLocation = compassLocation;
+    public static void setDynamicCompassLocation(Supplier<Location> compassSupplier) {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/CompassModel.java:63]: Why don't we use our new Poi interface and implement WaypointPoi as a "special" POI? That way you can get rid of this supplier weirdness and directly support it in the class.

**Review by magicus — COMMENTED** [review]
I just skimmed through it, found nothing that looks out of the ordinary, but I have not done a full review (no time for that, sorry).

**Pull request merged**